### PR TITLE
Set the correct VM family on the currently supported GPU instance types

### PR DESCRIPTION
**Pull request by bennerv**
### Which issue this PR addresses:

Fixes [ADO15807536)](https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/15807536)

### What this PR does / why we need it:

Fixes the incorrect NCASv3_T4 family type for GPU instances during quota validations\

### Test plan for issue:

Try to create a cluster with Standard_NC16as_T4_v3 instance types and have it fail quota validation.

```bash
400: ResourceQuotaExceeded: : Resource quota of Standard NCASv3_T4 Family exceeded. Maximum allowed: 0, Current in use: 0, Additional requested: 192.
```

### Is there any documentation that needs to be updated for this PR?

no

### CODE REVIEW

**Comment by bennerv** [comment]
Hitting the bug resolved in #2457.  Waiting for that to merge then e2e should be green.

**Review by SudoBrendan — DISMISSED** [review]
LGTM, pending item mentioned above/e2e

**Review by darthhexx — APPROVED** [review]
Validated using az CLI and LGTM.

**Pull request merged**
